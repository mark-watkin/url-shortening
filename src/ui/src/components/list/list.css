.list {
    display: flex;
    flex-direction: column;
}

.list-item {
    padding: 20px;
    overflow-wrap: break-word;
}

.list > .list-item:not(:last-child) {
    border-bottom: solid #D7D7D7;
    border-width: 1px;
}

.list-item-primary {
    font-weight: 400;
    font-size: 1rem;
    line-height: 1.5;
    color: #2a2a2a;
}

.list-item-secondary {
    font-weight: 400;
    font-size: 0.875rem;
    line-height: 1.43;
    color: #717070;
}