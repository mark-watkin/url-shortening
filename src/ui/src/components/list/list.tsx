import './list.css';

export interface ListItemProps {
    primaryText: string;
    secondaryText: string;
}

export interface ListProps {
    children?: React.ReactNode;
}

export const List: React.FC<ListProps> = ({children}) =>  {
    return <div className="list">
        {children}
    </div>
}

export const ListItem: React.FC<ListItemProps> = ({primaryText, secondaryText}) =>  {
    return <div className="list-item">
        <div className="list-item-primary">{primaryText}</div>
        <div className="list-item-secondary">{secondaryText}</div>    
    </div>
}