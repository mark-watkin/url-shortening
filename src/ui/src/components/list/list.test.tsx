import renderer from 'react-test-renderer';
import { List, ListItem } from './list';

it('renders correctly', () => {
  const tree = renderer
    .create(<List>
        <ListItem primaryText="Primary" secondaryText="Secondary" />
    </List>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});