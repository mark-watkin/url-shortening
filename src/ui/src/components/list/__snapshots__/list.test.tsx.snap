// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="list"
>
  <div
    className="list-item"
  >
    <div
      className="list-item-primary"
    >
      Primary
    </div>
    <div
      className="list-item-secondary"
    >
      Secondary
    </div>
  </div>
</div>
`;
